mineinabyss-features/src/main/kotlin/com/mineinabyss/helpers/HelperFunctions.kt

```diff
@@ -53,4 +55,33 @@ fun Player.updateBalance() {
     }
 }
 
+fun Player.bossbarCompass(loc: Location, bar: BossBar) {
+    val player = player ?: return
 
+    val dir = loc.clone().subtract(player.location).toVector()
+    val angleDir = (atan2(dir.z,dir.x) / 2 / Math.PI * 360 + 360) % 360
+    val angleLook = (atan2(player.location.direction.z,player.location.direction.x) / 2 / Math.PI * 360 + 360) % 360
+
+    when ((angleDir - angleLook + 180) % 360) {
+        in 0.0..22.5 -> bar.name(Component.text(":arrow_s:"))
```

Comment:
Instead of manually doing a ton of checks, just divide angle by 22.5, round down, then access the proper index in an array" `listOf(":arrow_sse:", ...)`